src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import os
+import re
+import socket
+
+from avocado.utils import process
+from control_test_base import ControlTestBase
+
+
+class NetDev(object):
+    # pylint: disable=too-few-public-methods
+    """A class to represent the information of a network device"""
+    def __init__(self, host=None, f_iface=None, providers=None, numa=None):
+        """Initialize the network device data object."""
+        self.host = host
+        self.f_iface = f_iface
+        self.providers = providers
+        self.numa = numa
+
+    def __repr__(self):
+        """Overwrite to display formated devices."""
+        self.__str__()
+
+    def __str__(self):
+        """Overwrite to display formated devices."""
+        return "\n".join("{}: {}".format(key, getattr(self, key, "MISSING"))
+                         for key in self.__dict__)
+
+    def __ne__(self, other):
+        """Override the default not-equal implementation."""
+        return not self.__eq__(other)
+
+    def __eq__(self, other):
+        """Override the default implementation to compare devices."""
+        status = isinstance(other, NetDev)
+        for key in self.__dict__:
+            if not status:
+                break
+            try:
+                status &= getattr(self, key) == getattr(other, key)
+            except AttributeError:
+                status = False
+        return status
+
+    def set_dev_info(self, prefix):
+        """Get all of this devices' information.
+
+        Args:
+            prefix (str): prefix path pointing to daos intall path
+        """
+        self.set_numa()
+        self.set_providers(prefix)
+
+    def set_numa(self):
```

Comment:
I like this.  I'm glad you found a good way to get the NUMA node data without using hwloc.